nncf/experimental/torch/fx/transformations.py

```diff
@@ -529,9 +583,217 @@ def fuse_conv_bn(model: torch.fx.GraphModule) -> None:
+
+    return pattern_per_channel, replacement_graph_per_channel
+
+
+def _get_pattern_replacement_per_tensor() -> (
+    Tuple[Callable[[torch.Tensor, torch.Tensor, torch.Tensor, int, int, torch.dtype], torch.Tensor]]
+):
+    """
+    Returns the patter and replacement function for the subgraph rewriter to
+    match and replace for per_tensor quantization
+    """
+
+    def pattern_per_tensor(weight, scale, zero_point, low, high, dtype):
+        quantized = torch.ops.quantized_decomposed.quantize_per_tensor.default(
+            weight, scale, zero_point, low, high, dtype
+        )
+        dequantized = torch.ops.quantized_decomposed.dequantize_per_tensor.default(
+            quantized, scale, zero_point, low, high, dtype
+        )
+        return dequantized
+
+    def replacement_graph_per_tensor(weight, scale, zero_point, low, high, dtype):
+        return torch.ops.aten.sub.Tensor(torch.ops.aten.mul.Tensor(weight, scale), zero_point)
+
+    return pattern_per_tensor, replacement_graph_per_tensor
+
+
+def _set_meta_for_matches(model: torch.fx.GraphModule, matches: torch.fx.subgraph_rewriter.ReplacedPatterns) -> None:
+    """
+    Sets the meta for the nodes which were matched by the subgraph_rewriter pattern.
+
+    :param model: Model whose nodes' meta are to be set.
+    :param matches: matches returned by the subgraph_rewriter
+    """
+    for match in matches:
+        mul_node = match.replacements[0]
+        sub_node = match.replacements[1]
+        _set_new_node_meta(mul_node, mul_node.args, torch.mul, model)
+        _set_new_node_meta(sub_node, sub_node.args, torch.sub, model)
+
+
+def _remove_constant_qdq_transformation(model: torch.fx.GraphModule) -> None:
+    """
+    Removes the Quantize/De-Quantize nodes for weight nodes by matching the pattern to be like follows:
+    weight scale zp
+        \   |    /
+         Quantize
+            |
+        Dequantize
+
+    and replace with the following:
+     weight scale
+       \     /
+       Multiply
+
+    :param model: Model to apply transformations to.
+    """
+
+    def match_filters(match, original_graph, graph):
+        for node in match.nodes_map:
+            if node.name == "weight" and match.nodes_map[node].op == "get_attr":
+                return True
+        return False
+
+    pattern, replacement = _get_pattern_replacement_per_channel()
+    matches_per_channel = torch.fx.subgraph_rewriter.replace_pattern_with_filters(
+        model, pattern, replacement, [match_filters]
+    )
+    _compress_qdq_constant_transformation(model, matches_per_channel)
+    _set_meta_for_matches(model, matches_per_channel)
+    pattern, replacement = _get_pattern_replacement_per_tensor()
+    matches_per_tensor = torch.fx.subgraph_rewriter.replace_pattern_with_filters(
+        model, pattern, replacement, [match_filters]
+    )
+    _compress_qdq_constant_transformation(model, matches_per_tensor)
+    _set_meta_for_matches(model.graph, matches_per_tensor)
```

Comment:
Let's reduce amount of identical lines
```suggestion
    patterns = [_get_pattern_replacement_per_tensor(), _get_pattern_replacement_per_channel()]
    for pattern, replacement in patterns:
        ...
```

Reply:
Done